src/content/developers/tutorials/how-to-mint-an-nft.md

```diff
@@ -0,0 +1,275 @@
+
+In your project home directory run:
+
+    npm install @alch/alchemy-web3
+
+## Step 2: Create a mint-nft.js file
+
+Inside your scripts directory, create an mint-nft.js file and add the following lines of code:
+
+    require('dotenv').config();
+    const API_URL = process.env.API_URL;
+    const { createAlchemyWeb3 } = require("@alch/alchemy-web3");
+    const web3 = createAlchemyWeb3(API_URL);
+
+## Step 3: Grab your contract ABI
+
+Our contract ABI (Application Binary Interface) is the interface to interact with our smart contract. You can learn more about Contract ABIs [here](https://docs.alchemyapi.io/alchemy/guides/eth_getlogs#what-are-ab-is). Hardhat automatically generates an ABI for us and saves it in the MyNFT.json file. In order to use this we’ll need to parse out the contents by adding the following lines of code to our mint-nft.js file:
+
+    const contract = require("../artifacts/contracts/MyNFT.sol/MyNFT.json");
+
+If you want to see the ABI you can print it to your console:
+
+    console.log(JSON.stringify(contract.abi));
+
+To run mint-nft.jsand see your ABI printed to the console navigate to your terminal and run
+
+    node scripts/mint-nft.js
+
+## Step 4: Configure the metadata for your NFT using IPFS
+
+If you remember from our tutorial in Part 1, our mintNFT smart contract function takes in a tokenURI parameter that should resolve to a JSON document describing the NFT's metadata— which is really what brings the NFT to life, allowing it to have configurable properties, such as a name, description, image, and other attributes.
+> *Interplanetary File System (IPFS) is a decentralized protocol and peer-to-peer network for storing and sharing data in a distributed file system.*
+
+We will use Pinata, a convenient IPFS API and toolkit, to store our NFT asset and metadata to ensure our NFT is truly decentralized. If you don’t have a Pinata account, sign up for a free account [here](https://pinata.cloud/signup) and complete the steps to verify your email.
+
+Once you’ve created an account:
+
+* Navigate to the “Pinata Upload” button on the top right
+
+* Upload an image to pinata — this will be the image asset for your NFT. Feel free to name the asset whatever you wish
+
+* After you upload, at the top of the page, there should be a green popup that allows you to view the hash of your upload → Copy that hashcode. You can view your upload at: [https://gateway.pinata.cloud/ipfs/<](https://gateway.pinata.cloud/ipfs/QmarPqdEuzh5RsWpyH2hZ3qSXBCzC5RyK3ZHnFkAsk7u2f)hash-code>
+
+For the more visual learners, the steps above are summarized here:
+
+![How to upload your image to Pinata](https://cdn-images-1.medium.com/max/2000/0*V_-3PGxnfsnLTGo4.gif)
```

Comment:
(reminder to download external images and place into same folder as markdown file, then reference locally using `![alt text for gif](./file-name.gif)`)